### DISCUSSION TITLE
How do I use plugins with pyenv?

### DISCUSSION THREAD

**Discussion by minaminao** [Vector35/binaryninja-api#3451] [Q&A] [answered]
How do I specify python of pyenv in Binary Ninja?

Even after executing `pip install -r requirements.txt` with some plugins, it seems that python of pyenv is not specified in Binary Ninja and the following error shows up in Log: `ModuleNotFoundError: No module named 'foo'`.

![image](https://user-images.githubusercontent.com/20497787/187799532-bf4129cb-3250-4819-a49d-45da0b5ac34a.png)

I believe that setting up them will solve this problem, but what exactly should I do?

**Comment by psifertex** [comment, marked as answer]
Use the drop down in the second setting shown above to select whatever version of python your `pip install` corresponds to. _Usually_ it's as simple as `which python` to see where that python is installed and matching the same version. If you're using one of the supported operating systems you shouldn't have to manually point to the exact file but should be able to select from a list.

Then just restart BN and it should have all the modules available.

**Reply by psifertex** [reply]
Also, if you've installed those python requirements into a sub-folder via pyenv, you can use `python -m site` to find that folder and add it to the third-option. The first setting on that list above is only needed for having the BN plugin manager install the dependencies automatically for you and you just point it at the python binary itself (the list on the second will be the shared libraries, dll, dylib, etc that corresponds to the right python)

**Reply by minaminao** [reply]
Thank you! It works with the following settings

![image](https://user-images.githubusercontent.com/20497787/187805003-b7f991f2-d25e-400f-8b4a-270b607244e9.png)

**Reply by psifertex** [reply]
Great! Glad that got it working.
